src/languages/scala.js

```diff
@@ -107,11 +107,22 @@ export default function(hljs) {
   const METHOD = {
     className: 'function',
     beginKeywords: 'def',
-    end: /[:={\[(\n;]/,
+    end: /(?=[:={\[(\n;])/,
```

Comment:
Please import `regex.js` tools and use `regex.lookahead` here for clarity.

Reply:
Now I'm only using the negative-lookahead. Should I define `regex.negativelookahead`?

Reply:
Huh, this looks positive?   I didn't flag any places you were using negative (or if I did please ignore)... I'm not sure we want to have tons of JS instead of regex but I have found the super common functions we already have more readable, such as positive `lookahead`, etc..

Reply:
I got mixed up with the negative-lookahead below (`/\s+(?!\))/`). Now I use `regex.lookahead` here. Sorry for the confusion.

Reply:
No problem.